pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -2620,6 +2639,12 @@ public ByteBufPair newMessageAndIntercept(long consumerId, long ledgerId, long e
             val brokerInterceptor = getBrokerService().getInterceptor();
             if (brokerInterceptor != null) {
                 brokerInterceptor.onPulsarCommand(command, this);
+
+                CompletableFuture<Consumer> consumerFuture = consumers.get(consumerId);
+                if (consumerFuture != null && consumerFuture.isDone() && !consumerFuture.isCompletedExceptionally()) {
+                    Consumer consumer = consumerFuture.getNow(null);
+                    brokerInterceptor.messageDispatched(this, consumer, ledgerId, entryId, metadataAndPayload);
```

Comment:
The `metadataAndPayload` does not contain the redeliveryCount, ackSet etc. I think it might be used in the future for the interceptor.

Reply:
Got it. Right now this is used to trace when messages are dispatched to consumers. Redeliveries will invoke this method again, right?

Reply:
Yes, redeliveries will invoke this method again, with the redelivery count, we are able to know this is message redelivery or not. It's ok for me.